### PR TITLE
Log errors when parsing configuration values

### PR DESCRIPTION
**Pull request by xlucas**
### What does this PR do?

Log a warning message every time a parsing error appears while retrieving a configuration parameter.

### Motivation

For now, these errors are silently ignored potentially leading to unclear situations for the user. We want to make sure there's an easier way to detect them.

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Review by gbbr — COMMENTED** [review]
It'd be best not to use `master` for pinning down `github.com/DataDog/viper` (for obvious reasons). Can we create a tag?

**Comment by gbbr** [comment]
I also wish there were better documentation added to the [changes in Viper](https://github.com/DataDog/viper/compare/v1.4.0...master) because it's unclear to me what type of parsing errors are possibly being returned and what the motivation behind these changes is. Since it was approved already it's all good, but wanted to comment on that.

**Comment by gbbr** [comment]
Please ignore my previous comment, after looking more deeply it's clearer to me now what those errors are.

**Comment by xlucas** [comment]
@gbbr Yes I believe it makes sense to draft a new version given the changes introduced.

File: Gopkg.toml [hunk 1]
```diff
@@ -128,7 +128,7 @@
 
 [[constraint]]
   name = "github.com/DataDog/viper"
-  version = "=v1.4.0"
+  branch = "master"
```
[the remark this answers is not in the document]
Comment (xlucas) [reply Gopkg.toml]: Yeah, pinning to a specific commit sounds like a better solution if we don't draft a new release. Given the changes in viper are pretty fresh, I would rather opt for this solution at first. WDYT @gbbr ?

As for the source override, I have no idea why it hasn't been used in that case. However, I feel like a better place for such a change, if it has to happen, would be in a separate PR.

[the remark this answers is not in the document]
Comment (gbbr) [reply Gopkg.toml]: > Yeah, pinning to a specific commit sounds like a better solution if we don't draft a new release. Given the changes in viper are pretty fresh, I would rather opt for this solution at first. WDYT @gbbr ?

Sure! I'm fine with anything as long as we don't pin to the tip of an active development branch.

**Review by clamoriniere — APPROVED** [review]

**Review by xvello — APPROVED** [review]
LGTM!

**Review by gbbr — APPROVED** [review]

**Review by remeh — APPROVED** [review]
LGTM.
I'm wondering, do we have any strategy regarding releases of our fork of Viper?

**Comment by xlucas** [comment]
@remeh Thanks! AFAIK nothing very precise for now.

**Pull request merged**